pkg/utils/utils.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// RevParse parses a git revision and returns a SHA1 on success, otherwise an
+// error.
+func RevParse(rev, workDir string) (string, error) {
+	// We hav eto use rev-list here to find the correct SHA1 for tags
+	output, err := ExecCmd("git rev-list -n1 "+rev, workDir)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(output), nil
+}
+
+// CloneTempRepository creates a temp directory containing the provided
+// GitHub repository via owner and name. It returns that directory if cloning
+// of the repository was successful, otherwise an error.
+func CloneTempRepository(owner, name string) (string, error) {
+	dir, err := ioutil.TempDir("", "release-notes")
+	if err != nil {
+		return "", err
+	}
+	if _, err := ExecCmd(
+		fmt.Sprintf("git clone https://github.com/%s/%s %s", owner, name, dir),
```

Comment:
What are your thoughts on using a golang-git or golang-github library? 

The use of said library(ies) is something I'm already planning to do so the `release-notes` tool can automatically generate a PR against k-sigs/release-notes when generating JSON. 

It's also cleaner and I feel like we should try to get away from shelling out as we start automating more.

Reply:
My main thought was about keeping the dependency chain low, but I would totally fine in using a library for this kind of work. I will follow up on that and check which would be a good fit. I also thought about having something like "find the latest tag", that we don't have to select it manually any more...

Reply:
Changed the implementation to rely on [go-git](https://github.com/src-d/go-git)